syntax = "proto3";
package v1;

message Response {
    float value = 1;
}
enum Error {
    NoFaces = 0;
    MoreThanOneFace = 1;
    UnautherizedFace = 2;
  }
message User {
    int64 id = 1;
    string first_name = 2;
    string last_name = 3;
    string user_name = 4;
    string profile_photo = 5;
}
message AuthenticatedResponse {
    bool authenticated = 1;
    Error message = 2;
    string error_code = 3;
}
message ValidateRequest{
    string api = 1;//API Version
    
    string face = 2;//Base 64 Encoded photo of face

}
message ValidateResponse{
    string api = 1;//API Version
    AuthenticatedResponse authenticated = 2;
    User user = 3;

}

service FaceRec {
    rpc AuthenticateFace(ValidateRequest) returns (ValidateResponse);
}